src/Component/Product/ProductProviderInterface.php

```diff
@@ -61,6 +61,11 @@ public function getProductCollectionManager();
      */
     public function getBaseControllerName();
 
+    /**
+     * @return string
+     */
+    public function getTemplatesPath();
```

Comment:
Please remove the docblock above or better, add a description, and use a return type hint.
```suggestion
    public function getTemplatesPath(): string;
```